### PR TITLE
refactor test_issue_237

### PR DESCRIPTION
**Pull request by jacadzaca**
splitting as per #400. Any idea why the tests pass in #400 but not here?

### CODE REVIEW

**Comment by niccokunzmann** [comment]
So, I see [this](https://github.com/collective/icalendar/actions/runs/3090206021/jobs/4998711463#step:6:79):

```
AssertionError: assert datetime.datetime(2017, 5, 11, 13, 30) == 
                       datetime.datetime(2017, 5, 11, 13, 30, tzinfo=zoneinfo.ZoneInfo(key='America/Sao_Paulo'))
```
That is an okay fail - but why is there no time zone in it?

Hm. Maybe a merge of master changes something?
Why would the event contain a time zone? There should be an error, I think.... The time zone has an unknown identifier.
Also: If it is a test refactoring - where are the tests that get deleted like in all the other PRs?

Maybe that helps.. let's find out together!

**Comment by jacadzaca** [comment]
The TZID property must be [specified before use](https://www.rfc-editor.org/rfc/rfc5545#section-3.2.19), so we need to parse a whole calendar. Now, we're back to the problem [with pytz](https://github.com/collective/icalendar/pull/400#issuecomment-1242119293), to which @niccokunzmann proposed [a solution](https://github.com/collective/icalendar/pull/400#discussion_r967359497) that I'm not sure how to implement

**Comment by niccokunzmann** [comment]
```
expected = datetime.datetime(2017, 5, 11, 13, 30, tzinfo=timezone('America/Sao_Paulo'))
```
I think, looking at the output:
```
assert datetime.datetime(2017, 5, 11, 13, 30, tzinfo=<DstTzInfo '(UTC-03:00) Brasília' Brasília standard-1 day, 21:00:00 STD>)
                     == datetime.datetime(2017, 5, 11, 13, 30, tzinfo=<DstTzInfo 'America/Sao_Paulo' LMT-1 day, 20:54:00 STD>)
                                                                                                                ^^^^^
```
This is the problem one has with pytz: You can not pass a time zone to the datetime. You must call `tz.localize` on the datetime object as far as I remember. There was an article about why to avoid pytz and this is one of the reasons - it does not (because old) obey how one would want to use the tz object nowadays. The offset is false because that was the offset before UTC alignment of the time zones. - all from memory this one is.

Could you try running `localize()` on this dt object?

**Comment by niccokunzmann** [comment]
I think, my comment there is that I would like to see that icalendar can actually choose a tz implementation and I would like to have all tests run across all different tz implementations.

I would pass it to `from_ical(..., timezone=pytz.tzinfo)`
Creating an issue ...

**Comment by niccokunzmann** [comment]
@jacadzaca can you merge master again into this one? I can not merge it, yet and I am too tired to resolve the conflict.

File: src/icalendar/tests/test_fixed_issues.py [hunk 1]
```diff
@@ -42,47 +42,3 @@ def test_issue_116(self):
...
-                    'END:VTIMEZONE',
-                    'BEGIN:VEVENT',
-                    'DTSTART;TZID=\"(UTC-03:00) Brasília\":20170511T133000',
-                    'DTEND;TZID=\"(UTC-03:00) Brasília\":20170511T140000',
-                    'END:VEVENT',
-                    'END:VCALENDAR',
-                    ]
-
-        cal = icalendar.Calendar.from_ical('\r\n'.join(ical_str))
-        self.assertEqual(cal.errors, [])
```
Comment (niccokunzmann) [inline src/icalendar/tests/test_fixed_issues.py:75]: We are missing this assertion. Apart from that: I trust that you copied it alright and the other assertions seem transferred.

**Review by niccokunzmann — CHANGES_REQUESTED** [review]

File: src/icalendar/tests/test_fixed_issues.py (hunk 1 above)
Comment (jacadzaca) [reply src/icalendar/tests/test_fixed_issues.py:75]: I readded the assertion in test_parses_event_with_non_ascii_tzid_issue_237

**Review by niccokunzmann — APPROVED** [review]

**Comment by niccokunzmann** [comment]
@jacadzaca Thanks for wroking together on this one!

**Pull request merged**